.eslintrc.js

```diff
@@ -0,0 +1,232 @@
+    "@typescript-eslint/unified-signatures": "error",
+    "comma-dangle": "off",
+    complexity: "off",
+    "constructor-super": "error",
+    curly: "off",
+    "dot-notation": "off",
+    eqeqeq: ["error", "smart"],
+    "guard-for-in": "error",
+    "id-denylist": "off",
+    "id-match": "off",
+    "import/order": [
+      "off",
+      {
+        alphabetize: {
+          caseInsensitive: true,
+          order: "asc",
+        },
+        "newlines-between": "ignore",
+        groups: [
+          ["builtin", "external", "internal", "unknown", "object", "type"],
+          "parent",
+          ["sibling", "index"],
+        ],
+        distinctGroup: false,
+        pathGroupsExcludedImportTypes: [],
+        pathGroups: [
+          {
+            pattern: "./",
+            patternOptions: {
+              nocomment: true,
+              dot: true,
+            },
+            group: "sibling",
+            position: "before",
+          },
+          {
+            pattern: ".",
+            patternOptions: {
+              nocomment: true,
+              dot: true,
+            },
+            group: "sibling",
+            position: "before",
+          },
+          {
+            pattern: "..",
+            patternOptions: {
+              nocomment: true,
+              dot: true,
+            },
+            group: "parent",
+            position: "before",
+          },
+          {
+            pattern: "../",
+            patternOptions: {
+              nocomment: true,
+              dot: true,
+            },
+            group: "parent",
+            position: "before",
+          },
+        ],
+      },
+    ],
+    "jsdoc/check-alignment": "error",
+    "jsdoc/check-indentation": "error",
+    "jsdoc/newline-after-description": "error",
+    "max-classes-per-file": "off",
+    "max-len": "off",
+    "new-parens": "error",
+    "no-bitwise": "error",
+    "no-caller": "error",
+    "no-cond-assign": "error",
+    "no-console": "off",
+    "no-debugger": "off",
+    "no-empty": "error",
+    "no-empty-function": "off",
+    "no-eval": "error",
+    "no-fallthrough": "off",
+    "no-invalid-this": "off",
+    "no-multiple-empty-lines": "off",
+    "no-new-wrappers": "error",
+    "no-shadow": "off",
+    "no-throw-literal": "error",
+    "no-trailing-spaces": "error",
+    "no-undef-init": "error",
+    "no-underscore-dangle": "off",
+    "no-unsafe-finally": "error",
+    "no-unused-expressions": "off",
+    "no-unused-labels": "error",
+    "no-use-before-define": "off",
+    "no-var": "error",
+    "object-shorthand": "error",
+    "one-var": ["error", "never"],
+    "prefer-arrow/prefer-arrow-functions": "off",
+    "prefer-const": "error",
+    quotes: "off",
+    radix: "error",
+    "space-before-function-paren": "off",
+    "spaced-comment": [
+      "error",
+      "always",
+      {
+        markers: ["/"],
+      },
+    ],
+    "use-isnan": "error",
+    "valid-typeof": "off",
+  },
+  overrides: [
+    {
```

Comment:
I suggest that we take all the javascript formatting out of the overrides and do the typescript formatting in the overrides.